### ISSUE TITLE
JsObject/JsValue does not preserve order of fields in version 2.6.11 and later

### ISSUE THREAD

**Issue by gyoho** [playframework/play-json#236]
### Play JSON Version >= 2.6.11

JsObject seems not preserving the fields order when removing a field. This seems occurring in `2.6.11` and later version. As I went through the commit, I found that the underlying collection type has been switched from `LinkedHashMap` provided by Scala library to Java's `LinedHashMap`. I suspect that commit caused this issue. (https://github.com/playframework/play-json/pull/193/files)

### API
- Scala version `2.12.8`
- SBT version `1.2.8`

### JDK
```
java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)
```

### Test Case

```scala
import play.api.libs.json._

case class Test(foo: Int, bar: String, baz: Boolean)
val testFmt: Format[Test] = Json.format[Test]
val test = Test(1, "bar", true)
val jsObj = testFmt.writes(test).as[JsObject]

val outOfOrder = jsObj - "baz"
```

#### Expected Behavior
```
outOfOrder should be {"foo":1,"bar":"bar"}
```

#### Actual Behavior
```
outOfOrder actually is {"bar":"bar","foo":1}
```

I couldn't find a pattern of the order, but apparently this happens when there are more than two fields in JsObject.

**Comment by levinson** [comment]
Good investigative work and thanks for opening the bug on play-json! I ran into this issue earlier when attempting to upgrade to a newer Play version and ended up backing out of the change.

**Comment by exp0nge** [comment]
@dwijnand What's the state of this? This is a breaking change.

**Comment by dwijnand** [comment]
Fixed in #253.
